src/main/java/org/apache/commons/lang3/time/DurationFormatUtils.java

```diff
@@ -221,72 +221,76 @@ public static String formatDurationWords(
+    public static String formatPeriod(final long startMillis, final long endMillis, final String format) {
         return formatPeriod(startMillis, endMillis, format, true, TimeZone.getDefault());
     }
 
     /**
      * <p>Formats the time gap as a string, using the specified format.
-     * Padding the left hand side of numbers with zeroes is optional and 
-     * the timezone may be specified. </p>
+     * Padding the left hand side of numbers with zeroes is optional and
+     * the time zone may be specified. </p>
      *
-     * <p>When calculating the difference between months/days, it chooses to 
-     * calculate months first. So when working out the number of months and 
-     * days between January 15th and March 10th, it choose 1 month and 
-     * 23 days gained by choosing January->February = 1 month and then 
-     * calculating days forwards, and not the 1 month and 26 days gained by 
-     * choosing March -> February = 1 month and then calculating days 
+     * <p>When calculating the difference between months/days, it chooses to
+     * calculate months first. So when working out the number of months and
+     * days between January 15th and March 10th, it choose 1 month and
+     * 23 days gained by choosing January-&gt;February = 1 month and then
+     * calculating days forwards, and not the 1 month and 26 days gained by
+     * choosing March -&gt; February = 1 month and then calculating days
      * backwards. </p>
      *
      * <p>For more control, the <a href="http://joda-time.sf.net/">Joda-Time</a>
      * library is recommended.</p>
-     * 
+     *
      * @param startMillis  the start of the duration
      * @param endMillis  the end of the duration
-     * @param format  the way in which to format the duration
-     * @param padWithZeros whether to pad the left hand side of numbers with 0's
-     * @param timezone the millis are defined in
-     * @return the time as a String
+     * @param format  the way in which to format the duration, not null
+     * @param padWithZeros  whether to pad the left hand side of numbers with 0's
+     * @param timezone  the millis are defined in
+     * @return the formatted duration, not null
+     * @throws java.lang.IllegalArgumentException if startMillis is greater than endMillis
      */
-    public static String formatPeriod(long startMillis, long endMillis, String format, boolean padWithZeros, 
-            TimeZone timezone) {
-
-        // Used to optimise for differences under 28 days and 
-        // called formatDuration(millis, format); however this did not work 
-        // over leap years. 
-        // TODO: Compare performance to see if anything was lost by 
-        // losing this optimisation. 
-        
-        Token[] tokens = lexx(format);
-
-        // timezones get funky around 0, so normalizing everything to GMT 
+    public static String formatPeriod(final long startMillis, final long endMillis, final String format, final boolean padWithZeros,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 23. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 388-405 into a separate method.